quiz/game_data.py

```diff
@@ -15,7 +17,8 @@ def get_correctly_answered_questions(self):
         return self.correctly_answered
 
     def register_correct_answer(self, question_id):
-        self.correctly_answered.add(question_id)
+        if Question.objects.get(pk=question_id).state == 'PUB':
+            self.correctly_answered.add(question_id)
```

Comment:
I don't understand how this fixes the issue, can you explain? As far as I understand, this only stops registering correct answers for questions that are not currently published? I think this is a very unlikely scenario, it would only happen if a question is retracted in the short time period between the question being displayed to the user and them answering it.

A much more likely scenario is that a user answers a published question, then (a few minutes or a few months later) that question is retracted. Now we should no longer count this question in the "You've answered 56 of 160 questions correctly." message, otherwise we risk the user answering all the questions on the site and seeing "You've answered 161 of 160 questions correctly".

Reply:
@knatten you're right. I'd thought only about "_a user follows a retracted question link, answers it => don't count that as a correct answer, done_" and hadn't thought about "_a user answers a published question, that question is then retracted_" (though it's clearly stated by the issue creator, sorry). For the first scenario there's a choice between patching `register_correct_answer()` and `get_correctly_answered_questions()`, and I chose the former, since it's called less often. Considering the second scenario, it seems you actually need to check for the retracted questions every `get_correctly_answered_questions()` invocation. I'll fix the PR accordingly.